drivers/clock_control/clock_control_rts5912_sccon.c

```diff
@@ -0,0 +1,247 @@
+
+static int rts5912_periph_clock_control(const struct device *dev, clock_control_subsys_t sub_system,
+					bool on_off)
+{
+	const struct rts5912_sccon_config *const config = dev->config;
+	struct rts5912_sccon_subsys *subsys = (struct rts5912_sccon_subsys *)sub_system;
+
+	SYSTEM_Type *sys_reg = (SYSTEM_Type *)config->reg_base;
+
+	uint32_t clk_grp = subsys->clk_grp;
+	uint32_t clk_idx = subsys->clk_idx;
+
+	uint32_t module_idx;
+
+	switch (clk_grp) {
+	case RTS5912_SCCON_I2C:
+		module_idx = (clk_idx - I2C0_CLKPWR) >> 2;
+
+		if (on_off) {
+			sys_reg->I2CCLK |= BIT(clk_idx);
+		} else {
+			sys_reg->I2CCLK &= ~BIT(clk_idx);
+		}
+
+		LOG_DBG("Turn I2C%d clock <%s>", module_idx, on_off ? "ON" : "OFF");
+		break;
+	case RTS5912_SCCON_UART:
+		if (on_off) {
+			sys_reg->UARTCLK = BIT(clk_idx);
+		} else {
+			sys_reg->UARTCLK &= ~BIT(clk_idx);
+		}
+
+		LOG_DBG("Turn UART0 clock <%s>", on_off ? "ON" : "OFF");
+		break;
+	case RTS5912_SCCON_ADC:
+		if (on_off) {
+			sys_reg->ADCCLK = BIT(clk_idx);
+		} else {
+			sys_reg->ADCCLK &= ~BIT(clk_idx);
+		}
+
+		LOG_DBG("Turn ADC clock <%s>", on_off ? "ON" : "OFF");
+		break;
+	case RTS5912_SCCON_PERIPH_GRP0:
+		if (on_off) {
+			sys_reg->PERICLKPWR0 |= BIT(clk_idx);
+		} else {
+			sys_reg->PERICLKPWR0 &= ~BIT(clk_idx);
+		}
+
+		LOG_DBG("Turn GRP0-%d clock <%s>", clk_idx, on_off ? "ON" : "OFF");
+		break;
+	case RTS5912_SCCON_PERIPH_GRP1:
+		if (on_off) {
+			sys_reg->PERICLKPWR1 |= BIT(clk_idx);
+		} else {
+			sys_reg->PERICLKPWR1 &= ~BIT(clk_idx);
+		}
+
+		LOG_DBG("Turn GRP1-%d clock <%s>", clk_idx, on_off ? "ON" : "OFF");
+		break;
+	case RTS5912_SCCON_PERIPH_GRP2:
+		if (on_off) {
+			sys_reg->PERICLKPWR2 |= BIT(clk_idx);
+		} else {
+			sys_reg->PERICLKPWR2 &= ~BIT(clk_idx);
+		}
+
+		LOG_DBG("Turn GRP2-%d clock <%s>", clk_idx, on_off ? "ON" : "OFF");
+		break;
+	case RTS5912_SCCON_SYS:
+		LOG_ERR("Not support peripheral group #%d-%d", clk_grp, clk_idx);
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Unknown peripheral group #%d", clk_grp);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int rts5912_clock_control_on(const struct device *dev, clock_control_subsys_t sub_system)
+{
+	return rts5912_periph_clock_control(dev, sub_system, true);
+}
+
+static int rts5912_clock_control_off(const struct device *dev, clock_control_subsys_t sub_system)
+{
+	return rts5912_periph_clock_control(dev, sub_system, false);
+}
+
+static int rts5912_clock_control_get_rate(const struct device *dev,
+					  clock_control_subsys_t sub_system, uint32_t *rate)
+{
+	const struct rts5912_sccon_config *const config = dev->config;
+	struct rts5912_sccon_subsys *subsys = (struct rts5912_sccon_subsys *)sub_system;
+
+	SYSTEM_Type *sys_reg = (SYSTEM_Type *)config->reg_base;
+
+	uint32_t clk_grp = subsys->clk_grp;
+	uint32_t clk_idx = subsys->clk_idx;
+	uint32_t module_idx;
+
+	uint32_t offset;
+	uint32_t src, divide, freq = 0;
+
+	switch (clk_grp) {
+	case RTS5912_SCCON_I2C:
+		module_idx = ((clk_idx - I2C0_CLKPWR) >> 2);
```

Comment:
nit: redundant parenthesis, drop

Reply:
removed it and have check the same problem in this .c file